src/libraries/System.Private.CoreLib/src/System/Boolean.cs

```diff
@@ -191,24 +187,14 @@ public int CompareTo(bool value)
 
         // Custom string compares for early application use by config switches, etc
         //
-        internal static bool IsTrueStringIgnoreCase(ReadOnlySpan<char> value)
-        {
-            return value.Length == 4 &&
-                    (value[0] == 't' || value[0] == 'T') &&
-                    (value[1] == 'r' || value[1] == 'R') &&
-                    (value[2] == 'u' || value[2] == 'U') &&
-                    (value[3] == 'e' || value[3] == 'E');
-        }
+        internal static bool IsTrueStringIgnoreCase(ReadOnlySpan<char> value) =>
+            value.Length == 4 &&
+            (BinaryPrimitives.ReadUInt64LittleEndian(MemoryMarshal.AsBytes(value)) | 0x0020002000200020) == 0x65007500720074; // "true" as a ulong, each char |'d with 0x0020 for case-insensitivity
 
-        internal static bool IsFalseStringIgnoreCase(ReadOnlySpan<char> value)
-        {
-            return value.Length == 5 &&
-                    (value[0] == 'f' || value[0] == 'F') &&
-                    (value[1] == 'a' || value[1] == 'A') &&
-                    (value[2] == 'l' || value[2] == 'L') &&
-                    (value[3] == 's' || value[3] == 'S') &&
-                    (value[4] == 'e' || value[4] == 'E');
-        }
+        internal static bool IsFalseStringIgnoreCase(ReadOnlySpan<char> value) =>
+            value.Length == 5 &&
+            (((BinaryPrimitives.ReadUInt64LittleEndian(MemoryMarshal.AsBytes(value)) | 0x0020002000200020) == 0x73006C00610066) & // "fals" as a ulong, each char |'d with 0x0020 for case-insensitivity
```

Comment:
why `&` not `&&` on line 196?

Reply:
To reduce a branch and make it more predictable in a situation where I expect expected-successful parsing to be more common than expected-unsuccessful.